spec/features/user_index_spec.rb

```diff
@@ -0,0 +1,28 @@
+require 'rails_helper'
+
+RSpec.describe 'User index page', type: :feature do
+  before :each do
+    @user = User.create(name: 'Debas', photo: 'image_link.jpg', bio: 'Developer from Macedonia')
+    @post = Post.create(author: @user, title: 'My title', text: 'My text')
+    @like = Like.create(author: @user, post_id: @post.id)
+
+    visit '/users'
+  end
+
+  it 'Can see the username.' do
+    expect(page).to have_content 'Debas'
+  end
+
+  it 'I can see the users profile picture' do
+    expect(page.find('img')['src']).to have_content 'image_link.jpg'
+  end
+
+  it 'Can see the number of posts the user has written.' do
+    expect(page).to have_content 'posts: 1'
+  end
+
+  it 'When I click on a user, I am redirected to that users show page' do
+    click_link(@user.id)
+    expect(current_path).to eq user_path @user.id
+  end
+end
```

Comment:
- [ ] As it is a  [requirement](https://github.com/microverseinc/curriculum-rails/blob/main/blog-app/projects/intergration_spec_and_n_1_project.md) for this project, "
`Login page:`

  - `I can see the username and password inputs and the "Submit" button.`
  - `When I click the submit button without filling in the username and the password, I get a detailed error.`
  - `When I click the submit button after filling in the username and the password with incorrect data, I get a detailed error.`
  - `When I click the submit button after filling in the username and the password with correct data, I am redirected to the root page.` " , please kindly create a new file and add the above mentioned tests. This [article](https://stackoverflow.com/questions/45198853/testing-sign-up-and-sign-in-with-capybara-rails-rspec-devise) should point you in the right direction.

Reply:
For this milestone, we are not supposed to implement the tests for the login page as we are on the new version of the RoR Module, where they missed the order of the milestones. That's why we are including the specs for the four views, not the login.

Reply:
@Mihndim2020, The Rails module is updated and the authentication project is after this one. The students don't have login pages yet so I think they can't create tests for the login page. The requirements of this project should be updated. 😄